src/client/components/pages/collection.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2020 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import EditionTable from './entities/edition-table';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import WorkTable from './entities/work-table';
+import request from 'superagent';
+
+
+const {Alert, Button, ButtonGroup, Row} = bootstrap;
+
+function getEntityTable(entityType) {
+	const tables = {
+		Edition: EditionTable,
+		Work: WorkTable
+	};
+	return tables[entityType];
+}
+
+function getEntityKey(entityType) {
+	const keys = {
+		Edition: 'editions',
+		Work: 'works'
+	};
+	return keys[entityType];
+}
+
+class CollectionPage extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			error: null,
+			selectedEntities: []
+		};
+
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleRemoveEntities = this.handleRemoveEntities.bind(this);
+	}
+
+	entityKey = getEntityKey(this.props.collection.entityType);
+
+	toggleRow(bbid) {
+		// eslint-disable-next-line react/no-access-state-in-setstate
+		const oldSelected = this.state.selectedEntities;
+		let newSelected;
+		if (oldSelected.find(selectedBBID => selectedBBID === bbid)) {
+			newSelected = oldSelected.filter(selectedBBID => selectedBBID !== bbid);
+		}
+		else {
+			newSelected = oldSelected.push(bbid) && oldSelected;
+		}
+		this.setState({
+			selectedEntities: newSelected
+		});
+	}
+
+	handleRemoveEntities() {
+		if (this.state.selectedEntities.length) {
+			const bbids = this.state.selectedEntities;
+			const submissionUrl = `/collection/${this.props.collection.id}/remove`;
+			request.post(submissionUrl)
+				.send({bbids})
+				.then((res) => {
+					window.location.href = `/collection/${this.props.collection.id}`;
+				}, (error) => {
+					this.setState({error: 'Internal Error'});
+				});
+		}
+		else {
+			this.setState({error: `No ${this.entityKey} selected`});
+		}
+	}
+
+	render() {
+		const errorComponent = this.state.error ?
+			<Alert bsStyle="danger">{this.state.error}</Alert> : null;
+		const EntityTable = getEntityTable(this.props.collection.entityType);
+		const propsForTable = {
+			[this.entityKey]: this.props.entities,
```

Comment:
That's a smart way to solve the issue of each entity table requiring a specific named key. Nice!